### ISSUE TITLE
Plans for default config improvements

### ISSUE THREAD

**Issue by zordsdavini** [qtile/qtile#1582]
Brainstorming what should be done to have friendly start for newcomers.

- add "Press <M-r> to spawn". Maybe in different color
- detect terminal (not all has xterm)
- review default keybindings (looks like some of them are not working)
- review default layouts, uncommenting should give working layout
- add default keybindings picture to documentation
- add optional dependencies to documentation
- start terminal by launching default Qtile

**Comment by pawamoy** [comment]
I was thinking about adding a shortcut to pop a notification to display the current key bindings and their description.

**Comment by zordsdavini** [comment]
I think we should compare to other WM what they have in default. For exampe DWM comes with almost nothing :) But sucless dwm tries to follow unix filosofy. In our case Qtile is not only WM. In my case it has many integrations and looks more like DE then WM :rofl: 
but @pawamoy I totally agree with you about notifications on failing parts. I remember you did some tricks for keybindings. We should add it. The case was notification app if I am not wrong. Just add fallback to logger.warning()

**Comment by am01k01h3** [comment]
I think it doesn't matter what other WMs do. I assume Our goal is for most people to have a lower barrier of entry to adopt qtile. If yes, then we should do whatever possible to make it easier for people.

For start, it would be great if there is a config.py present in the XDG config dir.
And, if the file has lots of options baked in, even if they are commented out, that would be awesome. So, anyone that bothered to try out qtile, at least has some way to move forward.

On the other hand, if the goal is similar to suckless to keep it elitist with higher barrier of entry, then so be it :+1:

**Comment by zordsdavini** [comment]
> I was thinking about adding a shortcut to pop a notification to display the current key bindings and their description.
@pawamoy in `qshell`  we have nice command `display_kb`. I tried to run from console `qtile-cmd -o cmd -f display_kb` but it gives very ugly print out.

**Comment by zordsdavini** [comment]
@am01k01h3 default config is already copied to users directory on first startup. I don't know about stable Qtile version but dev version has it

**Comment by zordsdavini** [comment]
for picture of default keybindings could be similar to ![xmonad](https://wiki.haskell.org/wikiupload/b/b8/Xmbindings.png)

**Comment by zordsdavini** [comment]
And about default layouts, I'd suggest switch to Max and Columns. Columns has same functionality as Stack but more posibilitites

**Comment by ramnes** [comment]
It's being proposed in #1628 that we had descriptions to Key objects, and I like the idea.

**Comment by arczan1** [comment]
I don't know if we should add description for Keys used to switch groups e.g.
`        Key([mod], i.name, lazy.group[i.name].toscreen(),
            desc="Switch to group {}".format(i.name))`
What do you think about that?

**Comment by ramnes** [comment]
Why not?

**Comment by arczan1** [comment]
At first I thought description should help reading code too but now I realized that it's what comments do. So I add it to config too.

**Comment by zordsdavini** [comment]
I was working with image generator, almost done and the part with description is not working (I thought that it works :D)
![mod4-control__](https://user-images.githubusercontent.com/4191548/79510783-b3278880-8046-11ea-8aa7-ef7a33d413e1.png)
![mod4-shift__](https://user-images.githubusercontent.com/4191548/79510787-b458b580-8046-11ea-9f16-f3cfb95dccb1.png)
![mod4__](https://user-images.githubusercontent.com/4191548/79510790-b458b580-8046-11ea-9d87-2db4d11271b1.png)

**Comment by arczan1** [comment]
#1630 
I created a pull request but I don't know how to mark it here.
Sorry for problem but this is my first pull request ever.

**Comment by zordsdavini** [comment]
Added PR https://github.com/qtile/qtile/pull/1631 to generate keybindings images. 
Now we need to decide what layouts should be default. May suggestion is max and columns. Do you agree @tych0 @ramnes @flacjacket ? 
Then we need to update default config for these layouts, fix documentation (if needed). We can use generator for images (maybe build should generate and copy images?

**Comment by zordsdavini** [comment]
for terminal detection it would be the easiest way just use bash `xterm || st || urxvt || <other popular terminals>`. Maybe there is alternative in python for such command?

**Comment by m-col** [comment]
On the topic of displaying keybindings, I think #1329 is quite a good idea and might play this role too. The advantage of this over something like an image is it will change when keybindings are changed and is immediately accessible.

And another idea for the default config... maybe a commented out demonstration of the wallpaper parameter to Screen, so that people are aware that it is there?

**Comment by zordsdavini** [comment]
> On the topic of displaying keybindings, I think #1329 is quite a good idea and might play this role too. The advantage of this over something like an image is it will change when keybindings are changed and is immediately accessible.

that image generator is more for documentation. Ex. I've put it into my qtile config repository (and qtile-example it uses as submodule) to get easy knowledge what keybindings are. But #1329 is needful, too. Because as you said it is easy to check current config and it can be displayed as list. For now `qshell> display_kb()` gives something similar, but I can't bind it or see from outside qshell in nice way.

> And another idea for the default config... maybe a commented out demonstration of the wallpaper parameter to Screen, so that people are aware that it is there?

I agree. But as I remember it depends on feh?

**Comment by ramnes** [comment]
No, it does not! I think we should put a default wallpaper actually.

**Comment by m-col** [comment]
The idea of a default wallpaper, the keybinding images generated by @zordsdavini, and #1595 together suggest that maybe there should be a super nice official Qtile colour scheme.

**Comment by zordsdavini** [comment]
Good idea @m-col . For now I'm using gruvbox everythere (grub, suckless tools, pywal to jetbrains products), but custom qtile style could give some uniqueness

**Comment by cr1ogen** [comment]
I will like me pywal integration

**Comment by m-col** [comment]
> I think we should put a default wallpaper actually.

TBH a plain black/dark grey background with the Qtile logo in the centre could be pretty nice default wallpaper.
